tests/common.go

```diff
@@ -2892,6 +2894,193 @@ func DeleteBucket(provider string, bucketName string) {
+					logrus.Error(err)
+					UpdateOutcome(event, err)
+					return
+				}
+
+				Step(
+					fmt.Sprintf("repl increase volume driver %s on app %s's volume: %v",
+						Inst().V.String(), ctx.App.Key, v),
+					func() {
+						logrus.Infof("Increasing repl with target node  [%v]", newReplID)
+						err = Inst().V.SetReplicationFactor(v, currRep+1, []string{newReplID}, false)
+						if err != nil {
+							logrus.Errorf("There is an error increasing repl [%v]", err.Error())
+							UpdateOutcome(event, err)
+						}
+					})
+
+				if err == nil {
+					Step(
+						fmt.Sprintf("reboot target node %s while repl increase is in-progres",
+							newReplNode.Hostname),
+						func() {
+
+							logrus.Info("Waiting for 10 seconds for re-sync to initialize before target node reboot")
+							time.Sleep(10 * time.Second)
+
+							err = Inst().N.RebootNode(newReplNode, node.RebootNodeOpts{
+								Force: true,
+								ConnectionOpts: node.ConnectionOpts{
+									Timeout:         1 * time.Minute,
+									TimeBeforeRetry: 5 * time.Second,
+								},
+							})
+							if err != nil {
+								logrus.Errorf("error rebooting node %v, Error: %v", newReplNode.Name, err)
+								UpdateOutcome(event, err)
+							}
+
+							err = validateReplFactorUpdate(v, currRep+1)
+							if err != nil {
+								err = fmt.Errorf("error in ha-increse after  target node reboot. Error: %v", err)
+								logrus.Error(err)
+								UpdateOutcome(event, err)
+							} else {
+								logrus.Infof("repl successfully increased to %d", currRep+1)
+							}
+						})
+				}
+			} else {
+				logrus.Error(err)
+				UpdateOutcome(event, err)
+
+			}
+		})
+}
+
+//HaIncreaseRebootSourceNode repl increase and reboot source node
+func HaIncreaseRebootSourceNode(event *EventRecord, ctx *scheduler.Context, v *volume.Volume, storageNodeMap map[string]node.Node) {
+	Step(
+		fmt.Sprintf("repl increase volume driver %s on app %s's volume: %v and reboot source node",
+			Inst().V.String(), ctx.App.Key, v),
+		func() {
+			currRep, err := Inst().V.GetReplicationFactor(v)
+			UpdateOutcome(event, err)
+
+			//if repl is 3 cannot increase repl for the volume
+			if currRep == 0 || currRep == 3 {
+				err = fmt.Errorf("cannot perform repl incease as current repl factor is %d", currRep)
+				logrus.Error(err)
+				UpdateOutcome(event, err)
+			}
+
+			if err == nil {
+				Step(
+					fmt.Sprintf("repl increase volume driver %s on app %s's volume: %v",
+						Inst().V.String(), ctx.App.Key, v),
+					func() {
+						replicaSets, err := Inst().V.GetReplicaSets(v)
+						if err == nil {
+							replicaNodes := replicaSets[0].Nodes
+							err = Inst().V.SetReplicationFactor(v, currRep+1, nil, false)
+							if err != nil {
+								logrus.Errorf("There is an error increasing repl [%v]", err.Error())
+								UpdateOutcome(event, err)
+							} else {
+								logrus.Info("Waiting for 30 seconds for re-sync to initialize before source nodes reboot")
```

Comment:
Same here ..

Reply:
checking for re-sync status will be little tricky in some cases where volume size is small  re-sync will happen very fast.
 And we are checking for volume is clean or not after ha-increase and reboot to verfiy succesfull ha-increase